src/vs/base/browser/ui/list/listWidget.ts

```diff
@@ -1865,7 +1865,7 @@ export class List<T> implements ISpliceable<T>, IDisposable {
 		return this.getFocus().map(i => this.view.element(i));
 	}
 
-	reveal(index: number, relativeTop?: number, paddingTop: number = 0): void {
+	reveal(index: number, relativeTop?: number, paddingTop: number = 0, applyRelativeTopOnlyIfNotVisible: boolean = false): void {
```

Comment:
There should not be a change to the list widget. The consumer of the list should be able to adapt the arguments depending on if an item is visible or not.

Reply:
Thank you for review. Agree the code now looks much nicer. Fixed.